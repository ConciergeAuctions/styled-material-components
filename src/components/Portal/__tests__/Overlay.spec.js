import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Overlay } from '../Overlay';

import {
  STYLED_COMPONENTS_CONTEXT,
  STYLED_COMPONENTS_CHILD_CONTEXT_TYPES,
} from '../../../../test/utils';

describe('Overlay', () => {
  let mounted;
  const onClick = jest.fn();
  it('mounts', () => {
    mounted = shallow(<Overlay onClick={onClick}>Content stuff!</Overlay>, {
      context: { ...STYLED_COMPONENTS_CONTEXT },
      childContextTypes: { ...STYLED_COMPONENTS_CHILD_CONTEXT_TYPES },
    });
    expect(mounted).toBeTruthy();
  });

  it('matches snapshot', () => {
    expect(toJSON(mounted)).toMatchSnapshot();
  });

  it.skip('renders its children as content', () => {});

  it.skip('fires onClick handler when the overlay is clicked', () => {});

  it.skip('does not fire onClick handler when the content is clicked', () => {});
});
