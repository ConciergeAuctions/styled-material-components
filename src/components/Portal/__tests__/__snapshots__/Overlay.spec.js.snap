// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay matches snapshot 1`] = `
.c0 {
  background: rgba(0,0,0,0.6);
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  opacity: 0;
  -webkit-transition: opacity 0.3s 0ms cubic-bezier(0,0,0.2,1);
  transition: opacity 0.3s 0ms cubic-bezier(0,0,0.2,1);
  will-change: opacity;
  pointer-events: none;
  contain: strict;
  overflow: hidden;
  z-index: 4;
}

<BaseOverlay
  className="c0"
  onClick={[MockFunction]}
>
  Content stuff!
</BaseOverlay>
`;
